components/service/glean/src/main/java/mozilla/components/service/glean/BooleanMetricType.kt

```diff
@@ -60,6 +65,8 @@ data class BooleanMetricType(
      */
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testHasValue(pingName: String = getStorageNames().first()): Boolean {
+        ioTask?.let { awaitJob(it) }
```

Comment:
Mh, what if we want to test that something was **not** written? Then await will throw, right? I think waiting should really be a separate operation to enable such test-cases.

**EDIT** - Woha. Reading the code again, looks like that test-case would still be possible, at least the first time the metric is used! That's because if nothing is written, then `ioTask` would be `null`. *However*, we're never resetting `ioTask` to `null`. But that is still not a problem, as we'd try to wait on a job that has already completed. The case I'm thinking of, is:

- `testHasValue` (with `ioTask = null`), should return `false`
- Record
- `testHasValue` (with `ioTask` as ongoing job), will return `true` after the job completes
- `testHasValue` (with `ioTask` as completed job), will return `true` immediately, because the job has already completed

Is my understanding correct? If so, I think we should at least mention in the docstring that this function waits for any outstanding recording operation to complete.

Reply:
I think that's a great idea to add the awaiting information to the docstring for the test functions, and yes, your understanding is correct.  If no one has called a metric function that uses `Dispatchers.API` which would assign the `Job` value to `ioTask`, then it will just skip the `awaitJob` and should return false with `testHasValue` or throw (as intended) with `testGetValue`.